src/reference/asciidoc/smb.adoc

```diff
@@ -171,6 +186,141 @@ public class SmbJavaApplication {
+
+This adapter produces message with payloads of type `InputStream`, letting files be fetched without writing to the local file system.
+Since the session remains open, the consuming application is responsible for closing the session when the file has been consumed.
+The session is provided in the `closeableResource` header (`IntegrationMessageHeaderAccessor.CLOSEABLE_RESOURCE`).
+Standard framework components, such as the `FileSplitter` and `StreamTransformer`, automatically close the session.
+See <<./file.adoc#file-splitter,File Splitter>> and <<./transformer.adoc#stream-transformer,Stream Transformer>> for more information about these components.
+The following example shows how to configure an `inbound-streaming-channel-adapter`:
+
+====
+[source, xml]
+----
+<int-smb:inbound-streaming-channel-adapter id="smbInbound"
+            channel="smbChannel"
+            session-factory="sessionFactory"
+            filename-pattern="*.txt"
+            filename-regex=".*\.txt"
+            filter="filter"
+            filter-expression="@myFilterBean.check(#root)"
+            remote-file-separator="/"
+            comparator="comparator"
+            max-fetch-size="1"
+            remote-directory-expression="'foo/bar'">
+        <int:poller fixed-rate="1000" />
+</int-smb:inbound-streaming-channel-adapter>
+----
+====
+
+Only one of `filename-pattern`, `filename-regex`, `filter`, or `filter-expression` is allowed.
+
+The `SmbStreamingMessageSource` adapter prevents duplicates for remote files with `SmbPersistentAcceptOnceFileListFilter` based on the in-memory `SimpleMetadataStore`.
+By default, this filter is also applied with the filename pattern (or regex).
+If you need to allow duplicates, you can use `AcceptAllFileListFilter`.
+Any other use cases can be handled by `CompositeFileListFilter` (or `ChainFileListFilter`).
+The Java configuration (<<smb-streaming-java,later in the document>>) shows one technique to remove the remote file after processing to avoid duplicates.
+
+For more information about the `SmbPersistentAcceptOnceFileListFilter`, and how it is used, see <<./file.adoc#remote-persistent-flf,Remote Persistent File List Filters>>.
+
+Use the `max-fetch-size` attribute to limit the number of files fetched on each poll when a fetch is necessary.
+Set it to `1` and use a persistent filter when running in a clustered environment.
+See <<smb-max-fetch>> for more information.
+
+The adapter puts the remote directory and file name in the `FileHeaders.REMOTE_DIRECTORY` and `FileHeaders.REMOTE_FILE` headers, respectively.
+The `FileHeaders.REMOTE_FILE_INFO` header provides additional remote file information (represented in JSON by default).
+If you set the `fileInfoJson` property on the `SmbStreamingMessageSource` to `false`, the header contains an `SmbFileInfo` object.
+The `SmbFile` object provided by the underlying Apache Net library can be accessed by using the `SmbFileInfo.getFileInfo()` method.
```

Comment:
I think this is a copy/paste artifact. 
I don't think that can treat our `jcifs` library as a `Apache Net`...